src/woo_publications/api/models.py

```diff
@@ -0,0 +1,69 @@
+import binascii
+import os
+
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from django_jsonform.models.fields import JSONField
+from phonenumber_field.modelfields import PhoneNumberField
+
+from .constants import PermissionOptions
+
+PERMISSION_SCHEMA = {
+    "type": "list",
+    "items": {
+        "type": "string",
+        "choices": [
+            {"title": label, "value": value}
+            for value, label in PermissionOptions.choices
+        ],
+        "widget": "multiselect",
+    },
+}
+
+
+class TokenAuth(models.Model):
+    token = models.CharField(_("token"), max_length=40, primary_key=True)
+    permissies = JSONField(
+        _("permission"),
+        help_text=_("The permission this API token has access to."),
+        schema=PERMISSION_SCHEMA,
+        blank=True,
+        null=True,
+    )
```

Comment:
`JSONField` is much too wide, this really should be an array field so we get proper database constraints to guarantee the structure of our data:

```py
PERMISSION_SCHEMA = {
    "type": "array",
    "items": {
        "type": "string",
        "choices": [
            {"title": label, "value": value}
            for value, label in PermissionOptions.choices
        ],
    },
}

...
    permissies = ArrayField(
        models.CharField(max_length=20, choices=PermissionOptions.choices),
        verbose_name=_("permission"),
        help_text=_("The permission this API token has access to."),
        schema=PERMISSION_SCHEMA,
        blank=True,
        default=list,
    )
```
this uses the `ArrayField` from `django-jsonform` 2.23+.